File: src/java/nz/co/searchwellington/repositories/TweetDAO.java
package nz.co.searchwellington.repositories;

import java.util.List;

import nz.co.searchwellington.model.Twit;

import org.hibernate.SessionFactory;
import org.hibernate.criterion.Expression;

public class TweetDAO {

	SessionFactory sessionFactory;

	public TweetDAO() {
	}

	public TweetDAO(SessionFactory sessionFactory) {
		this.sessionFactory = sessionFactory;
	}

	public Twit loadTweetByTwitterId(Long twitterId) {
		return (Twit) sessionFactory.getCurrentSession().createCriteria(
				Twit.class).add(Expression.eq("twitterid", twitterId))
				.setMaxResults(1).setCacheable(true).uniqueResult();
	}

	@SuppressWarnings("unchecked")
	public List<Twit> getAllTweets() {
		return sessionFactory.getCurrentSession().createCriteria(Twit.class)
				.list();
	}

	public void saveTwit(Twit twit) {
		sessionFactory.getCurrentSession().saveOrUpdate(twit);
		sessionFactory.getCurrentSession().flush();
	}

}

File: src/main/scala/nz/co/searchwellington/repositories/TweetDAO.scala
package nz.co.searchwellington.repositories

import nz.co.searchwellington.model.Twit
import nz.co.searchwellington.repositories.mongo.MongoRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component class TweetDAO @Autowired() (mongoRepository: MongoRepository) {

  def loadTweetByTwitterId(twitterId: Long): Twit = {
    // sessionFactory.getCurrentSession.createCriteria(classOf[Twit]).add(Restrictions.eq("twitterid", twitterId)).setMaxResults(1).setCacheable(true).uniqueResult.asInstanceOf[Twit]
    null // TODO
  }

  @SuppressWarnings(Array("unchecked")) def getAllTweets: Seq[Twit] = {
    // sessionFactory.getCurrentSession.createCriteria(classOf[Twit]).list.asInstanceOf[List[Twit]]
    Seq() //TODO
  }

  def saveTwit(twit: Twit) {
    // sessionFactory.getCurrentSession.saveOrUpdate(twit)
  }

}
